reconcile/utils/terrascript_client.py

```diff
@@ -342,17 +342,41 @@ def populate_iam_groups(self, roles):
     def _get_aws_username(user):
         return user.get('aws_username') or user['org_username']
 
+    @staticmethod
+    def _validate_mandatory_policies(account: dict,
+                                     user_policies: list[dict],
+                                     role_name: str) -> bool:
+        ok = True
+        mandatory_policies = \
+            [p for p in account.get('policies') or [] if p.get('mandatory')]
+        for mp in mandatory_policies:
+            if mp not in user_policies:
+                msg = \
+                    f"[{account['name']}] mandatory policy " + \
+                    f"{mp['name']} not associated to role {role_name}"
+                logging.error(msg)
+                ok = False
+        return ok
+
     def populate_iam_users(self, roles):
         for role in roles:
             users = role['users']
             if len(users) == 0:
                 continue
 
             aws_groups = role['aws_groups'] or []
+            user_policies = role['user_policies'] or []
+
             for aws_group in aws_groups:
                 group_name = aws_group['name']
-                account_name = aws_group['account']['name']
-                account_console_url = aws_group['account']['consoleUrl']
+                account = aws_group['account']
+                account_name = account['name']
+                account_console_url = account['consoleUrl']
+
+                ok = self._validate_mandatory_policies(
+                    account, user_policies, role['name'])
+                if not ok:
+                    raise ValueError('mandatory policies missing')
```

Comment:
optimization nit: instead of failing on the first group that does not attach all mandatory policies, this code could check all groups and fail afterwards if at least one did not satisfy the rule

feel free to ignore - consecutive PR fixes and PR checks will lead to clean configs as well

Reply:
i didn't ignore and see where that got us